lib/improver/utilities/ancillary_creation.py

```diff
@@ -28,12 +28,169 @@
+
+        return result
+
+    def difference_to_gradient(self, cube_x, cube_y):
+        """
+        This uses the grid spacing with difference in height fields
+        to calculate the gradient between grid spaces.
+
+        Args:
+            cube_x : iris.cube.Cube
+                The difference in height between adjacent grid squares
+                in the x direction.
+            cube_y: iris.cube.Cube
+                The difference in height between adjacent grid squares
+                in the y direction.
+
+         Returns:
+            gradient_x : iris.cube.Cube
+               A cube of the gradients based on orography in the x
+               direction.
+            gradient_y : iris.cube.Cube
+               A cube of the gradients based on orography in the y
+               direction.
+        """
+        grid_space_x = np.diff(cube_x.coord(axis='x').points)[0]
+        grid_space_y = np.diff(cube_y.coord(axis='y').points)[0]
+        gradient_x = cube_x.copy(data=abs(cube_x.data / grid_space_x))
+        gradient_y = cube_y.copy(data=abs(cube_y.data / grid_space_y))
+
+        return gradient_x, gradient_y
+
+    def normalise_cube(self, cubes, min_output_value=0, max_output_value=1):
+        """
+        This normalises a cube so that all of the numbers are between
+        min and max value which can be set.
+
+        Args:
+            cubes : iris.cube.Cubelist
+                A list of cubes that we need to take the cube_max and
+                cube_min from.
+            min_output_value : float
+                The minimum value we want our alpha to be
+            max_output_value : float
+                The maximum value we want our alpha to be
+
+        Returns:
+            normalised_cubes : iris.cube.Cube
+                A normalised cube based on the orography
+        """
+        cube_min = min([cube.data.min() for cube in cubes])
+        cube_max = max([cube.data.max() for cube in cubes])
+
+        normalised_cubes = iris.cube.CubeList()
+        for cube in cubes:
+            normalised_cube = cube.copy(data=(cube.data - cube_min) /
+                                        (cube_max - cube_min))
+            normalised_cube.data = (normalised_cube.data * (max_output_value
+                                    - min_output_value) + min_output_value)
+            normalised_cubes.append(normalised_cube)
+        return normalised_cubes
+
+    def scale_alpha_values(self, difference_cube):
+        """
+        This scales the alpha values depending on our equation
+        for alpha.
+
+        Args:
+            difference_cube : iris.cube.Cube
+                A cube of the normalised gradient
+
+        Returns:
+            difference_cube : iris.cube.Cube
+                The scaled cube of normalised gradient
+        """
+        difference_cube.data = (
+            self.coefficient * difference_cube.data**self.power +
+            self.intercept)
```

Comment:
This is quite complex given that default values for these parameters mean this function has no effect.  Are we envisaging a need for a scaling function of specifically this form?  If we're not sure about the form, should we include this at all?

Reply:
Remove intercept. Add an explanation to the doc string.